### PR TITLE
Duplicacy in user_program table for alumni users.

### PR DESCRIPTION
**Pull request by Satendra-SR**
#308 

### Types of changes

- [x] Bug fix (a change which fixes an issue)
- [ ] New feature (a change which adds functionality)

### Short description of what this resolves:

- Previously user_program creation and updation logic is working from the handle prompt flow with prompt_responses.
- Now the user_program creation and updation logic is shifted to the registration flow because we only need this while 
`On-Boarding Call`.
- `Upsert_user_program` logic is removed from `user_program` model because now we are directly calling `create` and `update` functions of the `user_program` model.

- We need this `user_program` creation and updation logic only with the `registration flow`.
- Now it resolves the problem that when the user_program is completed then the system is creating new user_program for alumni users. But, currently the system should have single entry per user in user_program table.

### Checklist:

- [x] I have performed a self-review of my own code.
- [x] The code follows the style guidelines of this project.
- [x] The code changes are passing the CI checks
- [ ] I have documented my code wherever required.
- [ ] The changes requires a change to the documentation.
- [ ] I have updated the documentation based on the my changes.

### CODE REVIEW

File: api/services/registration_service.py [hunk 1]
```diff
@@ -33,21 +33,42 @@ def handle_registration(self, jsonData):
...
                 if call_log:
                     registration_data = models.Registration.query.get_by_id(
                         call_log.registration_id
                     )
                     self.update_registration(registration_data, jsonData)
+                    user_program = (
+                        models.UserProgram.query.get_latest_active_user_program(
+                            self.user_id
+                        )
+                    )
```
Comment (GauravGusain98) [inline api/services/registration_service.py:47]: Lets move this outside the if condition. This way we don't have to write it in both `if` and `else` block

**Review by GauravGusain98 — APPROVED** [review]
Minor  suggestion. Else looks fine

File: api/services/registration_service.py (hunk 1 above)
Comment (Satendra-SR) [reply api/services/registration_service.py]: Done.

File: api/models/user_program.py [hunk 2]
```diff
@@ -5,30 +5,26 @@
...
     def create(self, user_id, data):
-        program_id = app.config["DEFAULT_PROGRAM_ID"]
-        preferred_time_slot = "AFTERNOON"
+        try:
+            program_id = app.config["DEFAULT_PROGRAM_ID"]
+            preferred_time_slot = app.config["DEFAULT_PROGRAM_TIME_CATEGORY"]
 
-        if data["program_id"]:
-            program_id = data["program_id"]
+            if data["program_id"]:
```
Comment (GauravGusain98) [inline api/models/user_program.py:13]:
```suggestion
            if data.get("program_id"):
```

**Pull request merged**